Modularize and clarify LB architecture & considerations (#2940)
theforeman/foreman-documentation: guides/common/modules/con_additional-maintenance-required-for-load-balancing-in-project.adoc

[id="additional-maintenance-required-for-load-balancing-in-project_{context}"]

> = Additional maintenance required for load balancing in {Project}

Reviewer: Just a general comment.

Whenever I see "in {Project}" or "for {Project}", I wonder whether it is necessary. Our module titles are very long, which makes them harder to scan. I think the context makes "in/for {Project}" unnecessary in most cases.

I am thinking about opening an issue to make our headers more concise. :-)

Author: @apinnick It's good that you're bringing this up. I lean towards more specific headings that include the product name to ensure reusable and context-free modules[1]. I know we don't have tooling to support proper reuse and searchability (yet!), but to me, that's a chicken-and-egg problem: Requests for that tooling are sometimes downplayed with the argument that module reuse is not that common, and trying to write content for reuse hits pushback with the argument that our tooling can't leverage it anyway :shrug: 

+1 on opening an issue to discuss this and reach some sort of agreement :)

[1] "Moreover, because modules are written as context-free elements independent of other modules, they are re-usable." as specified in the [mod docs guide](https://redhat-documentation.github.io/modular-docs/#modular-docs-terms-definitions).

Reviewer: I understand the logic behind including the product name in the header. At the same time, there does not appear to be a lot of module reuse. Wasn't that the reason we decided to remove the contexts? :-)

Reviewer 2: I can see both sides to this. However, I feel like this is out of the scope of this PR unless headers have been changed systematically. Can you please open an issue to track and discuss this @apinnick ?

Reviewer: Done . https://github.com/theforeman/foreman-documentation/issues/2956

Configuring {SmartProxies} to use a load balancer results in a more complex environment and requires additional maintenance.

The following additional steps are required for load balancing:

> * You must ensure that all {SmartProxies} have the same content views and synchronize all {SmartProxies} to the same content view versions

Reviewer: ```suggestion
* You must ensure that all {SmartProxies} have the same content views and synchronize all {SmartProxies} to the same content view versions.
```

> * You must upgrade each {SmartProxy} in sequence

Reviewer: ```suggestion
* You must upgrade each {SmartProxy} in sequence.
```

> * You must backup each {SmartProxy} that you configure regularly

Reviewer: ```suggestion
* You must back up each {SmartProxy} that you configure regularly.
```

The noun is "backup". The verb is "back up". It's like "login" (login credentials) vs "log in" (log in to the server).
